yabt/graph_test.py

```diff
@@ -335,3 +338,54 @@ def test_graph_dot_generation(basic_conf):
         assert '}' == dot_lines[-1]
         assert expected_dot_nodes == set(dot_lines[1:9])
         assert expected_dot_edges == set(dot_lines[9:18])
+
+
+@pytest.mark.usefixtures('in_simpleflat_project')
+def test_graph_dot_generation_colors(basic_conf):
+    build_context = BuildContext(basic_conf)
+    populate_targets_graph(build_context, basic_conf)
+    expected_targets = {':flask-0.10.1': TARGETS_COLORS['PythonPackage'],
+                        ':flask-hello-app': TARGETS_COLORS['Python']}
+    expected_dot_nodes = ['  "{}" [color="{}",];'.format(target, color)
+                          for target, color in expected_targets.items()]
+    with io.StringIO() as dot_io:
+        write_dot(build_context, basic_conf, dot_io)
+        dot_lines = dot_io.getvalue().strip('\n').split('\n')
+        for dot_node in expected_dot_nodes:
+            assert dot_node in dot_lines
+
+
+@pytest.mark.usefixtures('in_cpp_project')
+def test_no_buildenv_deps_in_dot(basic_conf):
+    build_context = BuildContext(basic_conf)
+    basic_conf.targets = ['hello:hello-app']
+    populate_targets_graph(build_context, basic_conf)
+    buildenv_targets = {':builder', ':ubuntu-gpg', ':clang', ':ubuntu',
+                        ':gnupg'}
+    expected_targets = {'hello:hello-app', 'hello:hello'}
+    with io.StringIO() as dot_io:
+        write_dot(build_context, basic_conf, dot_io)
+        all_targets = set(dot_io.getvalue().split('"'))
+        assert not buildenv_targets.intersection(all_targets)
```

Comment:
I think this makes the intention clearer:

```suggestion
        assert len(buildenv_targets.intersection(all_targets)) == 0
```

Reply:
I think it's a matter of taste, I like the first one better because you can read it like English - assert that there is no intersection